manim/mobject/text/numbers.py

```diff
@@ -327,10 +344,15 @@ def construct(self):
                 self.add(Integer(number=6.28).set_x(-1.5).set_y(-2).set_color(YELLOW).scale(1.4))
     """
 
-    def __init__(self, number=0, num_decimal_places=0, **kwargs):
+    def __init__(
+        self,
+        number: float | complex = 0,
```

Comment:
Does it make sense for an `Integer` to work with complex numbers?

Reply:
Not really. However, I thought that removing the option for `number` to be `complex` would violate the Liskov substitution principle by making an argument more specific, but it turns out MyPy is happy about that.